gravitee-am-identityprovider/gravitee-am-identityprovider-mongo/src/main/java/io/gravitee/am/identityprovider/mongo/authentication/MongoAuthenticationProvider.java

```diff
@@ -112,11 +113,17 @@ public Maybe<User> loadUserByUsername(Authentication authentication) {
                 })
                 .toList()
                 .flatMapMaybe(userEvaluations -> {
-                    if (userEvaluations.stream().filter(UserCredentialEvaluation::isPasswordValid).count() > 1) {
+                    final var validUsers = userEvaluations.stream().filter(UserCredentialEvaluation::isPasswordValid).collect(Collectors.toList());
+                    if (validUsers.size() > 1) {
                         LOGGER.debug("Authentication failed: multiple accounts with same credentials");
                         return Maybe.error(new BadCredentialsException("Bad credentials"));
                     }
+
                     var userEvaluation = userEvaluations.get(0);
+                    if (!validUsers.isEmpty()) {
+                        userEvaluation = validUsers.get(0);
```

Comment:
Suggestion: you can avoid reassigning `userEvaluation` by writing
`var userEvaluation = !validUsers.isEmpty() ? validUsers.get(0) : userEvaluations.get(0);`